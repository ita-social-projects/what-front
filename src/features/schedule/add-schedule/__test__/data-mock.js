const mockAllGroups = [
  { id: 212, name: 'SX-13' },
  { id: 214, name: 'SX-15' },
];

const mockAllActiveMentors = [
  { id: 7, firstName: 'Elizabeth', lastName: 'Hill' },
  { id: 8, firstName: 'Barbara', lastName: 'Harris' },
  { id: 9, firstName: 'Susan', lastName: 'Clark' },
];

const mockAllThemes = [
  { id: 26, name: '3D Printing' },
  { id: 24, name: 'Animation' },
  { id: 29, name: 'Buffer Overflow Exploits and Defenses' },
];

export const mockedData = {
  groups: {
    data: mockAllGroups,
    isLoading: false,
    isLoaded: false,
    error: '',
  },
  mentors: {
    data: mockAllActiveMentors,
    isLoading: false,
    isLoaded: false,
    error: '',
  },
  themes: {
    data: mockAllThemes,
    isLoading: false,
    isLoaded: false,
    error: '',
  },
};