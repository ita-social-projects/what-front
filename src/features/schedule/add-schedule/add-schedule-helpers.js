import { weekDays } from './add-schedule-data.js';

const findNameInObject = (arr, requiredParam) => (
  arr.find(({ name }) => name === requiredParam)
);

export const prepareToSubmit = (
  values,
  recurrencePatterns,
  allGroups,
  allThemes,
  allActiveMentors,
) => {
  const {
    patternType,
    interval,
    daysOfWeek,
    index,
    dates,
    startDate,
    finishDate,
    group,
    theme,
    mentor,
  } = values;

  const daysOfWeekForSend = daysOfWeek.map((el) => {
    if (el === 'Sunday') {
      return 0;
    }

    return weekDays.findIndex((item) => el === item) + 1;
  });

  const patternTypeforSend = recurrencePatterns.findIndex((item) => item === patternType);

  const groupID = findNameInObject(allGroups, group).id;
  const themeID = findNameInObject(allThemes, theme).id;
  const mentorID = allActiveMentors.find(({ firstName, lastName }) => `${firstName} ${lastName}` === mentor).id;

  const postData = {
    pattern: {
      type: patternTypeforSend,
      interval,
      daysOfWeek: daysOfWeekForSend,
      index,
      dates: [dates],
    },
    range: {
      startDate,
      finishDate,
    },
    context: {
      groupID,
      themeID,
      mentorID,
    },
  };

  return postData;
};

export const addDays = (days) => {
  const result = new Date();
  result.setDate(result.getDate() + days);

  return result;
};