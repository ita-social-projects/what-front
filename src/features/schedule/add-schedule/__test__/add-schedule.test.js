import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { Router } from 'react-router-dom';

import { AddSchedule } from '../add-schedule.js';
import { useFetching } from '../add-schedule-useFetch.js';
import { mockedData } from './data-mock.js';
import { recurrencePatterns } from '../add-schedule-data.js';
import { addDays } from '../add-schedule-helpers.js';

const historyMock = {
  push: jest.fn(),
  location: {},
  listen: jest.fn(),
  createHref: jest.fn(),
};

jest.mock('../add-schedule-useFetch', () => ({
  useFetching: jest.fn(),
}));

describe('test add-schedule component', () => {
  describe('add-schedule component', () => {
    it('should be rendered', () => {
      useFetching.mockReturnValue({ ...mockedData });

      const { getByTestId } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const addSchedule = getByTestId('add-schedule');

      expect(addSchedule).toMatchSnapshot();
    });
  });

  describe('add-schedule component has loader', () => {
    it('if areGroupsLoading && !areGroupsLoaded', () => {
      useFetching.mockReturnValue({ ...mockedData, groups: { isLoading: true } });

      const { container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const loader = container.querySelector('.spinner-border');

      expect(loader).toBeInTheDocument();
    });

    it('if areActiveMentorsLoading && !areActiveMentorsLoaded', () => {
      useFetching.mockReturnValue({ ...mockedData, mentors: { isLoading: true } });

      const { container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const loader = container.querySelector('.spinner-border');

      expect(loader).toBeInTheDocument();
    });

    it('if areThemesLoading && !areThemesLoaded', () => {
      useFetching.mockReturnValue({ ...mockedData, themes: { isLoading: true } });

      const { container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const loader = container.querySelector('.spinner-border');

      expect(loader).toBeInTheDocument();
    });
  });

  describe('add-schedule component has content', () => {
    beforeEach(() => {
      useFetching.mockReturnValue({
        ...mockedData,
        groups: { ...mockedData.groups, isLoaded: true },
        mentors: { ...mockedData.mentors, isLoaded: true },
        themes: { ...mockedData.themes, isLoaded: true },
        addSchedule: jest.fn(),
      });
    });

    it('first click on button "Add Schedule" right now after render is made disabled', async () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const buttonAdd = getByRole('button');

      await waitFor(() => {
        fireEvent.click(buttonAdd);
      });

      expect(buttonAdd).toBeDisabled();
    });

    it('Recurrence pattern is in the component', () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      expect(patternType).toBeInTheDocument();
    });

    it(`
      if Recurrence pattern is "Select a recurrence type",
      component should have seven times text "This field is required"`, async () => {
      const { getByRole, getAllByText } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'Select a recurrence type' } });
        const allRequiredFields = getAllByText('This field is required');
        expect(allRequiredFields).toHaveLength(7);
      });
    });

    it(`
      if Recurrence pattern is "Select a recurrence type",
      should disabled: DAYS OF WEEK, DAY OF MONTH, INDEX`, async () => {
      const { getAllByRole, getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'Select a recurrence type' } });

        const allCheckBoxes = getAllByRole('checkbox');

        allCheckBoxes.forEach((checkbox) => {
          expect(checkbox).toBeDisabled();
        });

        const dayOfMonth = getByRole('combobox', { name: 'Day of month:' });

        expect(dayOfMonth).toBeDisabled();

        const index = getByRole(
          'combobox',
          { name: 'Instance of the allowed days the event occurs:' },
        );

        expect(index).toBeDisabled();
      });
    });

    it(`
      if Recurrence pattern is "daily",
      component should have six times text "This field is required"`, async () => {
      const { getByRole, getAllByText } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        const allRequiredFields = getAllByText('This field is required');
        expect(allRequiredFields).toHaveLength(6);
      });
    });

    it(`
      if Recurrence pattern is "daily",
      should disabled: DAYS OF WEEK, DAY OF MONTH, INDEX`, async () => {
      const { getAllByRole, getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });

        const allCheckBoxes = getAllByRole('checkbox');

        allCheckBoxes.forEach((checkbox) => {
          expect(checkbox).toBeDisabled();
        });

        const dayOfMonth = getByRole('combobox', { name: 'Day of month:' });

        expect(dayOfMonth).toBeDisabled();

        const index = getByRole(
          'combobox',
          { name: 'Instance of the allowed days the event occurs:' },
        );

        expect(index).toBeDisabled();
      });
    });

    it(`
      if Recurrence pattern is "daily", should
      "Interval between each occurrence" have 31 items`, async () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        fireEvent.select(interval, { target: { value: '31' } });
        expect(interval.value).toBeTruthy();
        fireEvent.change(interval, { target: { value: '32' } });
        expect(interval.value).toBeFalsy();
      });
    });

    it(`
      if Recurrence pattern is "weekly",
      component should have six times text "This field is required"
      and one time text "Should be choosen at least one day"`, async () => {
      const { getByRole, getAllByText, getByText } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'weekly' } });
        const allRequiredFields = getAllByText('This field is required');
        expect(allRequiredFields).toHaveLength(6);

        const chooseDayOfWeek = getByText('Should be choosen at least one day');
        expect(chooseDayOfWeek).toBeTruthy();
      });
    });

    it(`
      if Recurrence pattern is "weekly",
      should disabled: DAY OF MONTH, INDEX,
      should enabled: DAYS OF WEEK`, async () => {
      const { getAllByRole, getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'weekly' } });

        const allCheckBoxes = getAllByRole('checkbox');

        allCheckBoxes.forEach((checkbox) => {
          expect(checkbox).toBeEnabled();
        });

        const dayOfMonth = getByRole('combobox', { name: 'Day of month:' });

        expect(dayOfMonth).toBeDisabled();

        const index = getByRole(
          'combobox',
          { name: 'Instance of the allowed days the event occurs:' },
        );

        expect(index).toBeDisabled();
      });
    });

    it(`
      if Recurrence pattern is "weekly", should
      "Interval between each occurrence" have 4 items`, async () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'weekly' } });
        fireEvent.change(interval, { target: { value: '4' } });
        expect(interval.value).toBeTruthy();

        fireEvent.change(interval, { target: { value: '5' } });
        expect(interval.value).toBeFalsy();
      });
    });

    it(`
      if Recurrence pattern is "on the same day of the month",
      component should have seven times text "This field is required"`, async () => {
      const { getByRole, getAllByText } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'on the same day of the month' } });
        const allRequiredFields = getAllByText('This field is required');
        expect(allRequiredFields).toHaveLength(7);
      });
    });

    it(`
      if Recurrence pattern is "on the same day of the month",
      should disabled: DAYS OF WEEK, INDEX,
      should enabled: DAY OF MONTH`, async () => {
      const { getAllByRole, getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'on the same day of the month' } });

        const allCheckBoxes = getAllByRole('checkbox');

        allCheckBoxes.forEach((checkbox) => {
          expect(checkbox).toBeDisabled();
        });

        const dayOfMonth = getByRole('combobox', { name: 'Day of month:' });

        expect(dayOfMonth).toBeEnabled();

        const index = getByRole(
          'combobox',
          { name: 'Instance of the allowed days the event occurs:' },
        );

        expect(index).toBeDisabled();
      });
    });

    it(`
      if Recurrence pattern is "on the same day of the month", should
      "Interval between each occurrence" have 12 items`, async () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'on the same day of the month' } });
        fireEvent.change(interval, { target: { value: '12' } });
        expect(interval.value).toBeTruthy();

        fireEvent.change(interval, { target: { value: '13' } });
        expect(interval.value).toBeFalsy();
      });
    });

    it(`
      if Recurrence pattern is "on the same day of the month", should
      "Day of month" have 31 items`, async () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Day of month:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'on the same day of the month' } });
        fireEvent.change(interval, { target: { value: '31' } });
        expect(interval.value).toBeTruthy();

        fireEvent.change(interval, { target: { value: '32' } });
        expect(interval.value).toBeFalsy();
      });
    });

    const SAME_DAY_IN_THE_MONTH = recurrencePatterns[3];

    it(`
      if Recurrence pattern is "on the same day of the week in the month",
      component should have seven times text "This field is required"
      and one time text "Should be choosen at least one day"`, async () => {
      const { getByRole, getAllByText, getByText } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: SAME_DAY_IN_THE_MONTH } });
        const allRequiredFields = getAllByText('This field is required');
        expect(allRequiredFields).toHaveLength(7);

        const chooseDayOfWeek = getByText('Should be choosen at least one day');
        expect(chooseDayOfWeek).toBeTruthy();
      });
    });

    it(`
      if Recurrence pattern is "on the same day of the week in the month",
      should disabled: DAY OF MONTH,
      should enabled: DAYS OF WEEK, INDEX`, async () => {
      const { getAllByRole, getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: SAME_DAY_IN_THE_MONTH } });

        const allCheckBoxes = getAllByRole('checkbox');

        allCheckBoxes.forEach((checkbox) => {
          expect(checkbox).toBeEnabled();
        });

        const dayOfMonth = getByRole('combobox', { name: 'Day of month:' });

        expect(dayOfMonth).toBeDisabled();

        const index = getByRole(
          'combobox',
          { name: 'Instance of the allowed days the event occurs:' },
        );

        expect(index).toBeEnabled();
      });
    });

    it(`
      if Recurrence pattern is "on the same day of the week in the month", should
      "Interval between each occurrence:" have 4 items
      and "Instance of the allowed days the event occurs:" has last, first items`, async () => {
      const { getByRole } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });
      const index = getByRole('combobox', { name: 'Instance of the allowed days the event occurs:' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: SAME_DAY_IN_THE_MONTH } });
        fireEvent.change(interval, { target: { value: '4' } });
        expect(interval.value).toBeTruthy();

        fireEvent.change(interval, { target: { value: '5' } });
        expect(interval.value).toBeFalsy();

        fireEvent.change(index, { target: { value: 'first' } });
        expect(index.value).toBeTruthy();

        fireEvent.change(index, { target: { value: 'last' } });
        expect(index.value).toBeTruthy();
      });
    });

    it(`
      if start date choosen later then now,
      should have field "Schedule's start date can't be in the past"`, async () => {
      const { getByText, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const startDate = container.querySelector('#startDate');

      await waitFor(() => {
        fireEvent.change(startDate, { target: { value: '2000-01-01T14:30' } });
        expect(getByText('Schedule`s start date can`t be in the past')).toBeTruthy();
      });
    });

    it(`
      if finish date choosen earlier then start date,
      should have field "Schedule's finish date can't be in the past"`, async () => {
      const { getByText, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const finishDate = container.querySelector('#finishDate');

      await waitFor(() => {
        fireEvent.change(finishDate, { target: { value: '2000-01-01T14:30' } });
        expect(getByText('Schedule`s finish date can`t be earlier start date')).toBeTruthy();
      });
    });

    it('if all fileds is filled, button "Add Schedule" is active', async () => {
      const { getByRole, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });
      const startDate = container.querySelector('#startDate');
      const finishDate = container.querySelector('#finishDate');
      const groups = getByRole('combobox', { name: 'Groups:' });
      const themes = getByRole('combobox', { name: 'Themes:' });
      const mentors = getByRole('combobox', { name: 'Mentors:' });
      const buttonAdd = getByRole('button');

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        fireEvent.change(interval, { target: { value: '31' } });
        fireEvent.change(startDate, { target: { value: addDays(1) } });
        fireEvent.change(finishDate, { target: { value: addDays(10) } });
        fireEvent.change(groups, { target: { value: 'SX-15' } });
        fireEvent.change(themes, { target: { value: '3D Printing' } });
        fireEvent.change(mentors, { target: { value: 'Susan Clark' } });

        expect(buttonAdd).toBeEnabled();
      });
    });

    it('if click button "Add Schedule" after all fields is filled, pop up "Confirm action"', async () => {
      const { getByRole, getByText, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });
      const startDate = container.querySelector('#startDate');
      const finishDate = container.querySelector('#finishDate');
      const groups = getByRole('combobox', { name: 'Groups:' });
      const themes = getByRole('combobox', { name: 'Themes:' });
      const mentors = getByRole('combobox', { name: 'Mentors:' });
      const buttonAdd = getByRole('button');

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        fireEvent.change(interval, { target: { value: '31' } });
        fireEvent.change(startDate, { target: { value: addDays(1) } });
        fireEvent.change(finishDate, { target: { value: addDays(10) } });
        fireEvent.change(groups, { target: { value: 'SX-15' } });
        fireEvent.change(themes, { target: { value: '3D Printing' } });
        fireEvent.change(mentors, { target: { value: 'Susan Clark' } });

        fireEvent.click(buttonAdd);

        expect(getByText('Are you sure you want to add this schedule?')).toBeTruthy();
      });
    });

    it('if click button "Close" in pop up "Confirm action", pop up is closed', async () => {
      const { getByRole, queryByRole, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });
      const startDate = container.querySelector('#startDate');
      const finishDate = container.querySelector('#finishDate');
      const groups = getByRole('combobox', { name: 'Groups:' });
      const themes = getByRole('combobox', { name: 'Themes:' });
      const mentors = getByRole('combobox', { name: 'Mentors:' });
      const buttonAdd = getByRole('button', { name: 'Add Schedule' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        fireEvent.change(interval, { target: { value: '31' } });
        fireEvent.change(startDate, { target: { value: addDays(1) } });
        fireEvent.change(finishDate, { target: { value: addDays(10) } });
        fireEvent.change(groups, { target: { value: 'SX-15' } });
        fireEvent.change(themes, { target: { value: '3D Printing' } });
        fireEvent.change(mentors, { target: { value: 'Susan Clark' } });

        fireEvent.click(buttonAdd);

        const cancel = queryByRole('button', { name: 'Cancel' });

        if (cancel) {
          fireEvent.click(cancel);
        }

        expect(cancel).not.toBeInTheDocument();
      });
    });

    it('if click button "x" in pop up "Confirm action", pop up is closed', async () => {
      const { getByRole, queryByRole, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });
      const startDate = container.querySelector('#startDate');
      const finishDate = container.querySelector('#finishDate');
      const groups = getByRole('combobox', { name: 'Groups:' });
      const themes = getByRole('combobox', { name: 'Themes:' });
      const mentors = getByRole('combobox', { name: 'Mentors:' });
      const buttonAdd = getByRole('button', { name: 'Add Schedule' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        fireEvent.change(interval, { target: { value: '31' } });
        fireEvent.change(startDate, { target: { value: addDays(1) } });
        fireEvent.change(finishDate, { target: { value: addDays(10) } });
        fireEvent.change(groups, { target: { value: 'SX-15' } });
        fireEvent.change(themes, { target: { value: '3D Printing' } });
        fireEvent.change(mentors, { target: { value: 'Susan Clark' } });

        fireEvent.click(buttonAdd);

        const close = queryByRole('button', { name: '×' });

        if (close) {
          fireEvent.click(close);
        }

        expect(close).not.toBeInTheDocument();
      });
    });

    it('if click button "Submit" in pop up "Confirm action", pop up is closed', async () => {
      const { getByRole, queryByRole, container } = render(
        <Router history={historyMock}>
          <AddSchedule />
        </Router>,
      );

      const patternType = getByRole('combobox', { name: 'Recurrence pattern:' });
      const interval = getByRole('combobox', { name: 'Interval between each occurrence:' });
      const startDate = container.querySelector('#startDate');
      const finishDate = container.querySelector('#finishDate');
      const groups = getByRole('combobox', { name: 'Groups:' });
      const themes = getByRole('combobox', { name: 'Themes:' });
      const mentors = getByRole('combobox', { name: 'Mentors:' });
      const buttonAdd = getByRole('button', { name: 'Add Schedule' });

      await waitFor(() => {
        fireEvent.change(patternType, { target: { value: 'daily' } });
        fireEvent.change(interval, { target: { value: '31' } });
        fireEvent.change(startDate, { target: { value: addDays(1) } });
        fireEvent.change(finishDate, { target: { value: addDays(10) } });
        fireEvent.change(groups, { target: { value: 'SX-15' } });
        fireEvent.change(themes, { target: { value: '3D Printing' } });
        fireEvent.change(mentors, { target: { value: 'Susan Clark' } });

        fireEvent.click(buttonAdd);

        const submit = queryByRole('button', { name: 'Submit' });

        if (submit) {
          fireEvent.click(submit);
        }

        expect(submit).not.toBeInTheDocument();
      });
    });
  });
});