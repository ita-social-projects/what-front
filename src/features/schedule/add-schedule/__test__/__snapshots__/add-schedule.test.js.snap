// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test add-schedule component add-schedule component should be rendered 1`] = `
<div
  data-testid="add-schedule"
>
  <form
    action="#"
  >
    <label
      class="row m-0 pt-3 d-flex"
      for="patternType"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Recurrence pattern:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          data-testid="patternType"
          id="patternType"
          name="patternType"
        >
          <option
            label="Select a recurrence type"
            value=""
          />
          <option
            label="daily"
            value="daily"
          />
          <option
            label="weekly"
            value="weekly"
          />
          <option
            label="on the same day of the month"
            value="on the same day of the month"
          />
          <option
            label="on the same day of the week in the month (example: \\"every second Wednesday of the month\\")"
            value="on the same day of the week in the month (example: \\"every second Wednesday of the month\\")"
          />
        </select>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      for="interval"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Interval between each occurrence:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          data-testid="interval"
          id="interval"
          name="interval"
        >
          <option
            label="Select an interval"
            value=""
          />
        </select>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      for="dayOfWeek"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Days of week:
      </p>
      <div
        class="col-md-8"
      >
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Monday"
          />
          <span>
            Monday
          </span>
        </span>
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Tuesday"
          />
          <span>
            Tuesday
          </span>
        </span>
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Wednesday"
          />
          <span>
            Wednesday
          </span>
        </span>
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Thursday"
          />
          <span>
            Thursday
          </span>
        </span>
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Friday"
          />
          <span>
            Friday
          </span>
        </span>
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Saturday"
          />
          <span>
            Saturday
          </span>
        </span>
        <span
          class="pr-3"
        >
          <input
            class="mr-1"
            disabled=""
            name="daysOfWeek"
            type="checkbox"
            value="Sunday"
          />
          <span>
            Sunday
          </span>
        </span>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="dates"
      for="dates"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Day of month:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          disabled=""
          id="dates"
          name="dates"
        >
          <option
            label="Select a day"
            value=""
          />
          <option
            label="1"
            value="1"
          />
          <option
            label="2"
            value="2"
          />
          <option
            label="3"
            value="3"
          />
          <option
            label="4"
            value="4"
          />
          <option
            label="5"
            value="5"
          />
          <option
            label="6"
            value="6"
          />
          <option
            label="7"
            value="7"
          />
          <option
            label="8"
            value="8"
          />
          <option
            label="9"
            value="9"
          />
          <option
            label="10"
            value="10"
          />
          <option
            label="11"
            value="11"
          />
          <option
            label="12"
            value="12"
          />
          <option
            label="13"
            value="13"
          />
          <option
            label="14"
            value="14"
          />
          <option
            label="15"
            value="15"
          />
          <option
            label="16"
            value="16"
          />
          <option
            label="17"
            value="17"
          />
          <option
            label="18"
            value="18"
          />
          <option
            label="19"
            value="19"
          />
          <option
            label="20"
            value="20"
          />
          <option
            label="21"
            value="21"
          />
          <option
            label="22"
            value="22"
          />
          <option
            label="23"
            value="23"
          />
          <option
            label="24"
            value="24"
          />
          <option
            label="25"
            value="25"
          />
          <option
            label="26"
            value="26"
          />
          <option
            label="27"
            value="27"
          />
          <option
            label="28"
            value="28"
          />
          <option
            label="29"
            value="29"
          />
          <option
            label="30"
            value="30"
          />
          <option
            label="31"
            value="31"
          />
        </select>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="index"
      for="index"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Instance of the allowed days the event occurs:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          disabled=""
          id="index"
          name="index"
        >
          <option
            label="Select in what days of week the event occurs"
            value=""
          />
          <option
            label="first"
            value="first"
          />
          <option
            label="second"
            value="second"
          />
          <option
            label="third"
            value="third"
          />
          <option
            label="fourth"
            value="fourth"
          />
          <option
            label="fifth"
            value="fifth"
          />
          <option
            label="sixth"
            value="sixth"
          />
          <option
            label="last"
            value="last"
          />
        </select>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="startDate"
      for="startDate"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Start date:
      </p>
      <div
        class="col-md-8"
      >
        <input
          class="w-100"
          id="startDate"
          name="startDate"
          type="datetime-local"
          value=""
        />
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="finishDate"
      for="finishDate"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        End date:
      </p>
      <div
        class="col-md-8"
      >
        <input
          class="w-100"
          id="finishDate"
          name="finishDate"
          type="datetime-local"
          value=""
        />
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="group"
      for="group"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Groups:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          id="group"
          name="group"
        >
          <option
            label="Select group"
            value=""
          />
          <option
            label="SX-13"
            value="SX-13"
          />
          <option
            label="SX-15"
            value="SX-15"
          />
        </select>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="theme"
      for="theme"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Themes:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          id="theme"
          name="theme"
        >
          <option
            label="Select theme"
            value=""
          />
          <option
            label="3D Printing"
            value="3D Printing"
          />
          <option
            label="Animation"
            value="Animation"
          />
          <option
            label="Buffer Overflow Exploits and Defenses"
            value="Buffer Overflow Exploits and Defenses"
          />
        </select>
      </div>
    </label>
    <label
      class="row m-0 pt-3 d-flex"
      data-testid="mentor"
      for="mentor"
    >
      <p
        class="col-md-4 font-weight-bolder"
      >
        Mentors:
      </p>
      <div
        class="col-md-8"
      >
        <select
          class="w-100"
          id="mentor"
          name="mentor"
        >
          <option
            label="Select mentor"
            value=""
          />
          <option
            label="Elizabeth Hill"
            value="Elizabeth Hill"
          />
          <option
            label="Barbara Harris"
            value="Barbara Harris"
          />
          <option
            label="Susan Clark"
            value="Susan Clark"
          />
        </select>
      </div>
    </label>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-3 col-4"
      >
        <button
          class="btn undefined"
          type="submit"
        >
          Add Schedule
        </button>
      </div>
    </div>
  </form>
</div>
`;
